File: project1/stop_words_remove.py
#!/usr/bin/env python

"""
A filter that _________.
"""

import fileinput

def process(line):
    """For each line of input, _____."""
    line = line[:-1]
    
    # read the stop-word-list.txt file
    file = open("stop-word-list.txt", "r")
    stopwords = file.read()

    # replace each stop words with None
    stop_list = stopwords.split("\r\n")
    for word in stop_list:
        if word == line:
            line = None
            
    print(line)

for line in fileinput.input():
    process(line)

File: project1/stop_words_remove.R
#!/usr/bin/env Rscript

stopwords <- read.table("stop-word-list.txt")

process <- function (input) {
  # Do something more here!
  if (input %in% stopwords[,1] == FALSE) {
    output <- input
  }
    else {output <- NULL}
  return(output)
}


f <- file("stdin")
open(f)
while(length(line <- readLines(f, n=1)) > 0) {
  line = process(line)
  write(line, stdout())
}
